### ISSUE TITLE
Auto-updating the field's value breaks when not using useForm's defaultValue

### ISSUE THREAD

**Issue by AndreSilva1993** [edmundhung/conform#771]
### Describe the bug and the expected behavior

Before https://github.com/edmundhung/conform/pull/729 went in I could have something like this:

```
const schema = z.object({ displayName: z.string() });

const [form, fields] = useForm({
  onValidate({ formData }) {
    return parseWithZod(formData, { schema });
  },
});

const defaultValueForInput = "defaultValueForInput";

<Texfield
  name={fields.displayName.name}
  defaultValue={defaultValueForInput}
/>
```

The input was properly initialised with the provided default value and that'd be it. With the https://github.com/edmundhung/conform/pull/729, either we provide a `defaultValue` to the `useForm` hook or the input's value will be overwritten to an empty text input / unchecked radio or checkbox.

I think such a change could either be marked as a **major** upgrade since the documentation / previous versions did not force users to provide the `defaultValue` to the useForm hook when wanting to pass a default value to their inputs - meaning that inputs that are initialised "out of the useForm realm" will be overwritten. Not necessarily arguing that this is better than using the `defaultValue` property in `useForm` since I can see arguments for both side, but it definitely caught me off-guard and broke my edit forms since they would be all empty by bumping to this version.

Or it could make sure that it only overwrites stuff if the `defaultValue` is provided and input is currently empty.

### Conform version

v1.2.1

### Steps to Reproduce the Bug or Issue

```
const schema = z.object({ displayName: z.string() });

const [fields] = useForm({ 
  onValidate: ({ formData }) => {
    return parseWithZod(formData, { schema });
  }
});

const inputDefaultValue = "default-value";

<input name={fields.displayName.name} defaultValue={inputDefaultValue} />
```

In here we can assert that the input will in fact be changed to an empty one.

### What browsers are you seeing the problem on?

Chrome, Firefox, Microsoft Edge, Safari

**Comment by itsmegood** [comment]
Same issue if my isActive boolean default value is set to true the form won't pick up the value until I turn the value off and then on again

**Comment by edmundhung** [comment]
Thanks for the writeup. I have gave it some thoughts last night and my current plan is to make this opt-in with an option to customize the behavior. I will put up a draft PR with more information later today. Sorry for the trouble!

**Comment by AndreSilva1993** [comment]
> Thanks for the writeup. I have gave it some thoughts last night and my current plan is to make this opt-in with an option to customize the behavior. I will put up a draft PR with more information later today. Sorry for the trouble!

That'd be perfect, thanks for the quick turnaround. It's really appreciated 👍

**Comment by edmundhung** [comment]
I have put up a fix on #775. Can you give the [pre-release](https://github.com/edmundhung/conform/pull/775#issuecomment-2356978130) a try and let me know if you run into any issues? Thanks.

**Comment by AndreSilva1993** [comment]
> I have put up a fix on #775. Can you give the [pre-release](https://github.com/edmundhung/conform/pull/775#issuecomment-2356979982) a try and let me know if you run into any issues? Thanks.

Tested this pre-release versions and no longer running into issues with my inputs being cleared due to the lack of `defaultValue`.

**Comment by edmundhung** [comment]
I'm sorry for the trouble. I've concluded that it's best to revert the relevant changes, and I've just published [v1.2.2](https://github.com/edmundhung/conform/releases/tag/v1.2.2).

**Comment by AndreSilva1993** [comment]
That works as well! Already tested `1.2.2` and can attest it fixes this issue, thus I'm closing this. Thanks!
